vars/buildPlugin.groovy

```diff
@@ -182,8 +182,6 @@ def call(Map params = [:]) {
                         }
                     }            
                 } finally {
-                    deleteDir()
-
                     if (hasDockerLabel()) {
```

Comment:
We could try this now that quite a few places are using ACI which I believe are one time agents? but previously we were re-using agents on every build and they were filling up constantly, (the docker prune change may be enough to fix this)

I'm ±0 it's easy to try out, but may bring back disk space issues

some other options would be:
```suggestion
                    try {
                      deleteDir()
                    } catch(ignored) {}
                    
                    if (hasDockerLabel()) {
```

```suggestion
                    if (isUnix()) {
                      deleteDir()
                    }
                    
                    if (hasDockerLabel()) {
```